src/main.rs

```
@@ -110,11 +131,29 @@ impl Op {
     fn print_root(&self, resolver: &Resolver) {
         match self {
             Op::Add(dep) | Op::Remove(dep) | Op::Update(_, dep) => {
-                let dir = resolver.dir(dep).map(|d| d.display().to_string()).unwrap_or_default();
+                let dir = resolver
+                    .dir(dep)
+                    .map(|d| d.display().to_string())
+                    .unwrap_or_default();
                 println!("Root of {}: {}", dep.name, dir);
             }
         }
     }
+    fn print_changelog(&self, resolver: &Resolver) {
+        match self {
+            Op::Update(old, new) => match (
+                dep_to_changelog(old, resolver),
+                dep_to_changelog(new, resolver),
+            ) {
+                (Ok(old), Ok(new)) => {
+                    let diff = changelog_diff(old, new);
+                    println!("{}", diff);
+                }
+                _ => (),
```

I'm pretty sure we don't want to silently ignore errors.